components/debounce/core/browser/debounce_rule.cc

```diff
@@ -52,6 +54,34 @@ std::string_view CanonicalizeHostForMatching(std::string_view host_piece) {
   return host_piece;
 }
 
+// Extract the host from |url| using a simple parsing algorithm
+// WARNING: this is a special-purpose function whose output should not be used.
+std::string NaivelyExtractHostnameFromUrl(const std::string& url) {
+  CHECK(GURL(url).SchemeIsHTTPOrHTTPS());
+
+  const std::string kHttp =
+      base::StrCat({url::kHttpScheme, url::kStandardSchemeSeparator});
+  const std::string kHttps =
+      base::StrCat({url::kHttpsScheme, url::kStandardSchemeSeparator});
+  std::string mutable_url = url;
+
+  if (base::StartsWith(mutable_url, kHttps,
+                       base::CompareCase::INSENSITIVE_ASCII)) {
+    mutable_url = mutable_url.substr(kHttps.length());
+  } else if (base::StartsWith(mutable_url, kHttp,
+                              base::CompareCase::INSENSITIVE_ASCII)) {
+    mutable_url = mutable_url.substr(kHttp.length());
+  }
+
+  const std::vector<std::string> parts = base::SplitString(
```

Comment:
```suggestion
  // Known limitation: this will not work properly with origins that include port numbers or with IPv6 hostnames.
  const std::vector<std::string> parts = base::SplitString(
```

Reply:
Added.